hop/io.py

```diff
@@ -111,11 +114,32 @@ def open(self, url, mode="r", metadata=False):
             raise ValueError("mode must be either 'w' or 'r'")
 
 
-class Deserializer(Enum):
-    CIRCULAR = models.GCNCircular
-    VOEVENT = models.VOEvent
-    BLOB = models.MessageBlob
+def _load_deserializer_plugins():
+    """Load all registered deserializer plugins.
+
+    """
+    # set up plugin manager
+    manager = pluggy.PluginManager("hop")
+    manager.add_hookspecs(plugins)
+
+    # load in base models
+    manager.register(models)
+
+    # load external models
+    try:
+        manager.load_setuptools_entrypoints("hop_plugin")
```

Comment:
Note for later - make sure to document how to define conda recipes that provide entry_points. I thought that it wouldn't be possible, but it turns out you can do it: https://stackoverflow.com/questions/50106303/how-to-i-create-an-entry-point-to-a-conda-package-application

Reply:
Will do. That's something I didn't quite get from defining an entry point via conda. If you set up entry points via setuptools in `setup.py`, `setup.cfg` or `pyproject.toml`, does that propagate in the right way through a conda recipe or do you need to also specify it explicitly?

Reply:
As an update, it seems like the answer to that question is yes. I'm looking at various feedstocks for pytest plugins for conda which use the same discovery mechanism and it looks like they don't define the entry points within the recipe in order for this to work out: https://github.com/conda-forge/pytest-console-scripts-feedstock/blob/master/recipe/meta.yaml